gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java

```diff
@@ -115,13 +131,22 @@ public void flush() throws InterruptedException {
     awaitAllOutstandingBatches();
   }
 
-  /** Sends accumulated elements asynchronously for batching. */
+  /**
+   * Sends accumulated elements asynchronously for batching.
+   *
+   * <p>Note: This method can be invoked concurrently unlike {@link #add} and {@link #close}, which
+   * can only be called from single user thread. Please take caution to avoid race condition.
```

Comment:
"a single"

Reply:
Thanks for the feedback; Have updated the Javadoc.